text/0028-netowrk-collections-and-variables.md

```diff
@@ -0,0 +1,94 @@
+* Feature Name: `network collections / variables revision`
+* Start Date: 2021-08-11
+* RFC PR: TBA
+* Issue: TBA0
+
+# Summary
+[summary]: #summary
+This RFC describes some upcoming revisions to the NetworkCollections (e.g. `NetworkSet`, `NetworkDictionary`, `NetworkList`) as well as the underlying `NetowrkVariable` system.
+
+# Motivation
+[motivation]: #motivation
+
+The current programming model allows for unsafe, race-condition-prone operations.  Its design also in its current form is not readily adaptable to future snapshot plans.  Lastly, the goals of the library to be a performant netcode system are in conflict with some of its lack of limitations
+
+
+# Guide-Level Explanation
+[guide-level]: #guid-level
+
+## Permission Changes
+
+Today, the library has 3 levels of permissions
+
+### Old: Read & Write Permissions
+
+* `ServerOnly`
+* `OwnerOnly`
+* `Everyone`
+
+
+### New: Read Permissions
+* `Everyone`
+* `ServerOnly`
+
+Going forward, Network Variables will only be (directly) writable by the server.  Previously, a client could change state on the server by writing to its local network variable instance.  As we move towards the snapshot tick-based model this free-form mechanism becomes intractable to support at least in its current form.  With this change, when a client wants to cause a state change it will need to use a different means such as sending an RPC.  
+
+Hence, with this release there are only *Read* permissions.  Either a variable can be read by everyone, or the specific owner of the variable.  The server can always read all variables.
+
```

Comment:
While I think it is pretty clear that the server is the only one with write permissions, is the reason owner clients can't write to a NetworkVariable due to the issue with change in ownership?